Source/WebCore/style/ElementRuleCollector.cpp

```diff
@@ -633,21 +646,23 @@ std::pair<bool, std::optional<Vector<ElementRuleCollector::ScopingRootWithDistan
         */
         auto isWithinScopingRootsAndScopeEnd = [&](const auto& selectorList) {
             auto match = [&] (const auto* scopingRoot, const auto* selector) {
+                auto subContext = context;
+                subContext.scope = scopingRoot;
                 const auto* ancestor = &element();
                 while (ancestor) {
+                    auto match = checker.match(*selector, *ancestor, subContext);
+                    if (match)
+                        return true;
                     if (ancestor == scopingRoot) {
-                        // The end of the scope has to be a descendant of the start of the scope.
+                        // The end of the scope can't be an ancestor of the start of the scope.
                         return false;
                     }
-                    auto match = checker.match(*selector, *ancestor, context);
-                    if (match)
-                        return true;
                     ancestor = ancestor->parentElement();
                 }
                 return false;
             };
 
-            for (auto [scopingRoot, distance] : scopingRoots) {
+            for (auto [scopingRoot, _] : scopingRoots) {
```

Comment:
fancy!

Reply:
Actually, it's just a regular identifier 😅 . In the future, it shall be more useful.

 https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2020/p2169r0.pdf